.github/workflows/pull-request.yml

```diff
@@ -114,16 +149,45 @@ jobs:
             runTsTests: false
 
     steps:
+      - name: Configure Change Filter
+        id: filter
+        shell: bash
+        run: |
+          FILTER_ANY_CHANGE="false"  # default value
+          if [[ "${{ needs.changes.outputs.github }}" == "true" || "${{ needs.changes.outputs.apps }}" == "true" || "${{ needs.changes.outputs.bindings }}" == "true" || "${{ needs.changes.outputs.cmake }}" == "true" || "${{ needs.changes.outputs.include }}" == "true" || "${{ needs.changes.outputs.libs }}" == "true" || "${{ needs.changes.outputs.root }}" == "true" || "${{ needs.changes.outputs.scripts }}" == "true" || "${{ needs.changes.outputs.source }}" == "source" || "${{ needs.changes.outputs.typescript }}" == "true" ]]; then
+            FILTER_ANY_CHANGE="true"
+          fi
+          FILTER_CPP_CHANGE="false"  # default value
+          if [[ "${{ needs.changes.outputs.cmake }}" == "true" || "${{ needs.changes.outputs.include }}" == "true" || "${{ needs.changes.outputs.libs }}" == "true" || "${{ needs.changes.outputs.source }}" == "source" ]]; then
```

Comment:
here's an example of setting a variable using arbitrary logic, in this case defining when the cpp code has changed and those steps need to run